### PR TITLE
add support for get guild bans options

### PR DESCRIPTION
**Pull request by lts20050703**
Closes: #2156 
Upstream: https://github.com/discord/discord-api-docs/commit/74f5787e83b12bd75ba7c451027c6de6b61ae431

### CODE REVIEW

File: helpers/guilds/getBans.ts [hunk 1]
```diff
@@ -4,8 +4,14 @@ import { DiscordBan } from "../../types/discord.ts";
...
 /** Returns a list of ban objects for the users banned from this guild. Requires the BAN_MEMBERS permission. */
-export async function getBans(bot: Bot, guildId: bigint) {
-  const results = await bot.rest.runMethod<DiscordBan[]>(bot.rest, "get", bot.constants.endpoints.GUILD_BANS(guildId));
+export async function getBans(bot: Bot, guildId: bigint, options: GetBans) {
+  const results = await bot.rest.runMethod<DiscordBan[]>(
+    bot.rest,
+    "get",
+    `${bot.constants.endpoints.GUILD_BANS(guildId)}?limit=${options.limit ?? 1000}&before=${
+      options.before ?? null
+    }&after=${options.after ?? null}`,
```
Comment (lts20050703) [inline helpers/guilds/getBans.ts:13]: @itohatweb @Skillz4Killz is this how I am supposed to do query string params?

Comment (itohatweb) [reply helpers/guilds/getBans.ts:13]: You can use an if statement

Comment (lts20050703) [reply helpers/guilds/getBans.ts:13]: alr then

File: helpers/guilds/getBans.ts [hunk 2]
```diff
@@ -4,8 +4,18 @@ import { DiscordBan } from "../../types/discord.ts";
...
+
+  if (options?.limit) queries.push(`limit=${options.limit}`);
+  if (options?.after) queries.push(`after=${options.after}`);
+  if (options?.before) queries.push(`before=${options.before}`);
+
+  const results = await bot.rest.runMethod<DiscordBan[]>(
+    bot.rest,
+    "get",
+    `${bot.constants.endpoints.GUILD_BANS(guildId)}${queries.length ? `?${queries.join("&")}` : ""}`,
+  );
```
Comment (itohatweb) [inline helpers/guilds/getBans.ts:18]:
```suggestion
  let url = bot.constants.endpoints.GUILD_BANS(guildId);

  if (options) {
    url += "?";

    if (options.limit) url += `limit=${options.limit}&`;
    if (options.after) url += `after=${options.after}&`;
    if (options.before) url += `before=${options.before}`;
  }

  const results = await bot.rest.runMethod<DiscordBan[]>(bot.rest, "get", url);
```

File: helpers/guilds/getBans.ts [hunk 3]
```diff
@@ -17,3 +27,12 @@ export async function getBans(bot: Bot, guildId: bigint) {
...
   );
 }
+
+export interface GetBans {
+  /** number of users to return (up to maximum 1000). Default: 1000 */
+  limit?: number;
+  /** consider only users before given user id */
+  before?: bigint;
+  /** consider only users after given user id */
+  after?: bigint;
```
Comment (itohatweb) [inline helpers/guilds/getBans.ts:37]:
```suggestion
  /** Number of users to return (up to maximum 1000). Default: 1000 */
  limit?: number;
  /** Consider only users before given user id */
  before?: bigint;
  /** Consider only users after given user id */
  after?: bigint;
```

**Review by itohatweb — CHANGES_REQUESTED** [review]

File: helpers/guilds/getBans.ts (hunk 2 above)
Comment (lts20050703) [reply helpers/guilds/getBans.ts:18]: @itohatweb
Is `url?` valid? (everything is optional)
Is `?&after` valid? (everything is optional)

Comment (itohatweb) [reply helpers/guilds/getBans.ts:18]: I discussed this design with skillz.
I proposed this way of doing and therefore yes it is valid.

**Review by itohatweb — APPROVED** [review]

**Review by Skillz4Killz — APPROVED** [review]

**Pull request merged**